### PR TITLE
[SYCLomatic] reduce_argmin and reduce_argmax 

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adding `dpct::reduce_argmin` and `dpct::reduce_argmax`

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc [hunk 1]
```diff
@@ -1775,6 +1775,36 @@ inline void segmented_sort_pairs(
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|reduce_argmax|dpct
+// DPCT_DEPENDENCY_EMPTY
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: reduce_argmin should depend on dpct::arg_index_input_iterator

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc [hunk 2]
```diff
@@ -1775,6 +1775,36 @@ inline void segmented_sort_pairs(
...
+  auto ret = ::std::max_element(
+       ::std::forward<_ExecutionPolicy>(policy), input_arg_idx,
+       input_arg_idx + n,
+       [](const auto &a, const auto &b) { return (a.value < b.value); });
+  ::std::copy(::std::forward<_ExecutionPolicy>(policy), ret, ret + 1, output);
+}
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|reduce_argmin|dpct
+// DPCT_DEPENDENCY_EMPTY
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: reduce_argmax should depend on dpct::arg_index_input_iterator

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc (hunk 1 above)
Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: Thanks, fixed.

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc (hunk 2 above)
Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: Thanks, fixed.

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**